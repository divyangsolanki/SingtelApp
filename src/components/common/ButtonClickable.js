import React from 'react';
import {Text, Button, View, TouchableOpacity} from 'react-native';

const ButtonClickableIOS = theProps => {
  const {textStyleLocal, buttonCommonStyle, textBody} = styles;

  const {onPress, children, textStyle, testID} = theProps.props;

  return (
    <View style={buttonCommonStyle}>
      <Text style={[textStyleLocal, textStyle, textBody]}>{children}</Text>
      <Button
        testID={testID}
        onPress={onPress}
        title={
          '                                \n                                \n                                '
        }
      />
    </View>
  );
};

const ButtonClickableAndroid = theProps => {
  const {textStyleLocal, buttonCommonStyle, textBody} = styles;

  const {onPress, children, textStyle, testID} = theProps.props;

  return (
    <TouchableOpacity onPress={onPress} testID={testID}>
      <View style={buttonCommonStyle}>
        <Text style={[textStyleLocal, textStyle, textBody]}>{children}</Text>
      </View>
    </TouchableOpacity>
  );
};

const ButtonClickable = theProps => {
  return global.isAndroid ? (
    <ButtonClickableAndroid props={theProps} />
  ) : (
    <ButtonClickableIOS props={theProps} />
  );
};

const styles = {
  textStyleLocal: {
    alignSelf: 'center',
    color: '#ffffff',
    fontSize: 14,
    letterSpacing: 0,
    fontStyle: 'normal',
    textAlign: 'center',
  },
  textBody: {
    position: 'absolute',
    width: 260,
  },
  buttonCommonStyle: {
    alignItems: 'center',
    justifyContent: 'center',
    width: 260,
    height: 64,
    borderRadius: 32,
    backgroundColor: '#003566',
  },
};

export default ButtonClickable;
