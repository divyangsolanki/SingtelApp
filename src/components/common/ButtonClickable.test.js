import React from 'react';
import {render} from '@testing-library/react-native';
import ButtonClickable from './ButtonClickable';

describe('Test ButtonClickable', () => {
  it('Button', () => {
    global.isAndroid = true;
    const wrapper = render(
      <ButtonClickable onPress={() => {}}>Restart</ButtonClickable>,
    );
    expect(wrapper.toJSON()).toMatchSnapshot();
  });

  // it('Button with enabled state with small size flag', () => {
  //   global.isAndroid = false;
  //   const wrapper = render(
  //     <ButtonClickable
  //       onPress={() => {}}
  //       isButtonDisabled={true}
  //       isSmallSizeButton={true}>
  //       SIGN IN
  //     </ButtonClickable>,
  //   );
  //   expect(wrapper.toJSON()).toMatchSnapshot();
  // });

  // it('Button with enabled state with small size flag = false', () => {
  //   global.isAndroid = false;
  //   const wrapper = render(
  //     <ButtonClickable
  //       onPress={() => {}}
  //       isButtonDisabled={true}
  //       isSmallSizeButton={false}>
  //       SIGN IN
  //     </ButtonClickable>,
  //   );
  //   expect(wrapper.toJSON()).toMatchSnapshot();
  // });

  // it('Button with disabled state', () => {
  //   global.isAndroid = true;
  //   const wrapper = render(
  //     <ButtonClickable onPress={() => {}} isButtonDisabled={true}>
  //       SIGN IN
  //     </ButtonClickable>,
  //   );
  //   expect(wrapper.toJSON()).toMatchSnapshot();
  // });
  it('Button with disabled state (isAndroid = false)', () => {
    global.isAndroid = false;
    const wrapper = render(
      <ButtonClickable onPress={() => {}} isButtonDisabled={true}>
        SIGN IN
      </ButtonClickable>,
    );
    expect(wrapper.toJSON()).toMatchSnapshot();
  });
});
