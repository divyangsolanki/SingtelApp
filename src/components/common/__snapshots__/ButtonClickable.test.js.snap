// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test ButtonClickable Button 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#003566",
        "borderRadius": 32,
        "height": 64,
        "justifyContent": "center",
        "width": 260,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "alignSelf": "center",
            "color": "#ffffff",
            "fontSize": 14,
            "fontStyle": "normal",
            "letterSpacing": 0,
            "textAlign": "center",
          },
          undefined,
          Object {
            "position": "absolute",
            "width": 260,
          },
        ]
      }
    >
      Restart
    </Text>
  </View>
</View>
`;

exports[`Test ButtonClickable Button with disabled state (isAndroid = false) 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#003566",
      "borderRadius": 32,
      "height": 64,
      "justifyContent": "center",
      "width": 260,
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "alignSelf": "center",
          "color": "#ffffff",
          "fontSize": 14,
          "fontStyle": "normal",
          "letterSpacing": 0,
          "textAlign": "center",
        },
        undefined,
        Object {
          "position": "absolute",
          "width": 260,
        },
      ]
    }
  >
    SIGN IN
  </Text>
  <View
    accessibilityRole="button"
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "margin": 8,
              "textAlign": "center",
            },
          ]
        }
      >
                                        
                                
                                
      </Text>
    </View>
  </View>
</View>
`;
